File: app-kotlin/src/main/java/com/tritiumstudios/phasmophobiaevidencetool/listeners/CompositeListener.java
package com.tritiumstudios.phasmophobiaevidencetool.listeners;

import android.view.View;

import java.util.ArrayList;

public class CompositeListener implements View.OnClickListener {

    private final ArrayList<View.OnClickListener> registeredListeners = new ArrayList<>();

    public void registerListener (View.OnClickListener listener) {
        registeredListeners.add(listener);
    }

    @Override
    public void onClick(View view) {
        for(View.OnClickListener listener:registeredListeners) {
            listener.onClick(view);
        }
    }
}

File: app-view/src/main/java/com/tritiumgaming/phasmophobiaevidencepicker/listeners/CompositeListener.kt
package com.tritiumgaming.phasmophobiaevidencepicker.listeners

import android.view.View

class CompositeListener : View.OnClickListener {
    private val registeredListeners = ArrayList<View.OnClickListener>()

    fun registerListener(listener: View.OnClickListener) {
        registeredListeners.add(listener)
    }

    override fun onClick(view: View) {
        for (listener in registeredListeners) {
            listener.onClick(view)
        }
    }
}